rdf-star-cg-spec.html

```diff
@@ -412,6 +425,18 @@ <h2>Grammar</h2>
             <a data-cite="SPARQL11-QUERY#rPropertyListPath">PropertyListPath</a>
           </td>
         </tr>
+        <tr id="rObjectListPath">
+          <td>[86]</td>
+          <td>`ObjectListPath`</td>
+          <td>::=</td>
+          <td>
+            <a data-cite="SPARQL11-QUERY#rObjectPath">ObjectPath</a> <a href="#rAnnotationPattern">AnnotationPattern</a>`?`
+            `(`
+              <code class="token">','</code>
+              <a data-cite="SPARQL11-QUERY#rObjectPath">ObjectPath</a> <a href="#rAnnotationPattern">AnnotationPattern</a>`?`
```

Comment:
This has to be a separate production.

One "AnnotationPattern" that allows paths inside it (and a grammar to rule not to allow compound paths) 
and one that does not have paths.

This parallels how triple patterns / path patterns work for WHERE (paths) and CONSTRUCT and update.

Reply:
Andy, I have added another commit which introduces such separate productions (commit https://github.com/w3c/rdf-star/pull/65/commits/802ff102aed787662af53ae6de712951dc701a1b). See the preview: 

Is that what you meant?

With the grammar extended in this way, it is now possible to write the following:
```
SELECT * WHERE {
   :s :p :o {| :pp1/:pp2 ?oo |}
}
```
...which can be expanded to:
```
SELECT * WHERE {
   :s :p :o .
   <<:s :p :o>> :pp1/:pp2 ?oo .
}
```
So, this works. However, I am not sure how intuitive it is for users to be able to write such annotation patterns with path expressions inside. On the other hand, the expansion step that replaces the annotation pattern does not become more complex by this extension. Hence, it doesn't hurt to have it, I guess.

Another thing that is possible with the grammar extended in this way is the following:
```
SELECT * WHERE {
   :s :p :o {| :pp [:ppp1/:ppp2 ?o] |}
}
```
...which can be expanded to:
```
SELECT * WHERE {
   :s :p :o .
   <<:s :p :o>> :pp1 [:ppp1/:ppp2 ?o]
}
```
So, this is also okay I think.

Is it these things that you had in mind?